cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -279,7 +278,7 @@ public List<RunCountResult> getProgramTotalRunCounts(List<ProgramReference> prog
       .collect(Collectors.toMap(RunCountResult::getProgramReference, c -> c));
 
     List<RunCountResult> result = new ArrayList<>();
-    for (ProgramReference programReference : programCounts.keySet()) {
+    for (ProgramReference programReference : programRefs) {
```

Comment:
This was a mistake in previous PRs, should loop through the passed in `ProgramReference` to return correct result in case it's not found